monitoring/monitoring_internal_test.go

```diff
@@ -181,528 +192,543 @@ func TestPeriodicReport(t *testing.T) {
+			UsedDisk:   2300,
+			InTraffic:  250,
+			OutTraffic: 150,
+		},
+		{
+			InstanceIdent: cloudprotocol.InstanceIdent{
+				ServiceID: "service1",
+				SubjectID: "subject2",
+				Instance:  2,
 			},
-		}); err != nil {
-		t.Fatalf("Can't start monitoring service: %s", err)
+			RAM:        2200,
+			CPU:        uint64(math.Round((45 + 45) / float64(runtime.NumCPU()))),
+			UsedDisk:   4600,
+			InTraffic:  150,
+			OutTraffic: 250,
+		},
+		{
+			InstanceIdent: cloudprotocol.InstanceIdent{
+				ServiceID: "service1",
+				SubjectID: "subject2",
+				Instance:  2,
+			},
+			RAM:        2200,
+			CPU:        uint64(math.Round((45 + 45) / float64(runtime.NumCPU()))),
+			UsedDisk:   2300,
+			InTraffic:  150,
+			OutTraffic: 250,
+		},
 	}
 
-	terminate := false
-
-	for terminate != true {
-		select {
-		case data := <-monitor.GetMonitoringDataChannel():
-			if len(data.ServiceMonitoring) != 2 {
-				t.Errorf("Wrong number of services: %d", len(data.ServiceMonitoring))
-			}
+	for i, item := range testData {
+		processesData = append(processesData, &testProcessData{
+			uid:       int32(item.monitoringConfig.UID),
+			quotaData: item.quotaData,
+		})
 
-			for resource, numAlerts := range alertMap {
-				if numAlerts != 2 {
-					t.Errorf("Wrong number of %s alerts: %d", resource, numAlerts)
-				}
-			}
+		instanceID := fmt.Sprintf("instance%d", i)
+		if err := monitor.StartInstanceMonitor(instanceID, item.monitoringConfig); err != nil {
+			t.Fatalf("Can't start monitoring service: %s", err)
+		}
 
-			if len(alertMap) != 5 {
-				t.Error("Not enough alerts")
-			}
+		instanceTrafficMonitoringData[instanceID] = item.trafficMonitoring
+		expectedInstanceAlertCount += item.alertExpectedCount
+	}
 
-			terminate = true
+	defer func() {
+		processesData = nil
+	}()
 
-		case <-time.After(sendDuration * 2):
-			t.Fatal("Monitoring data timeout")
+	select {
+	case monitoringData := <-monitor.GetMonitoringDataChannel():
+		if len(alertMap) != expectedInstanceAlertCount {
+			t.Error("Alerts should not be generated")
 		}
-	}
-
-	alertMap = make(map[string]int)
 
-	err = monitor.StopMonitorService("service1")
-	if err != nil {
-		t.Fatalf("Can't stop monitoring service: %s", err)
-	}
+		if len(monitoringData.ServiceInstances) != len(monitoringInstances) {
+			t.Fatalf("Incorrect instance monitoring count %d", len(monitoringData.ServiceInstances))
+		}
 
-	terminate = false
+		for _, receivedMonitoring := range monitoringData.ServiceInstances {
+			found := false
 
-	for terminate != true {
-		select {
-		case data := <-monitor.GetMonitoringDataChannel():
-			if len(data.ServiceMonitoring) != 1 {
-				t.Errorf("Wrong number of services: %d", len(data.ServiceMonitoring))
+			for _, expectedMonitoring := range monitoringInstances {
+				if expectedMonitoring == receivedMonitoring {
+					found = true
+					break
+				}
 			}
 
-			if len(alertMap) != 0 {
-				t.Error("Not enough alerts")
+			if !found {
```

Comment:
You can use continue with label in this case:
```golang
          monitoringLoop:
		for _, receivedMonitoring := range monitoringData.ServiceInstances {
			for _, expectedMonitoring := range monitoringInstances {
				if expectedMonitoring == receivedMonitoring {
					continue monitoringLoop
				}
			}

			t.Error("Unexpected monitoring data")
		}
```

Reply:
Done